packages/nextjs/utils/scaffold-eth/parseTxnValue.ts

```diff
@@ -0,0 +1,22 @@
+import { BigNumber } from "ethers";
+
+/**
+ * @dev utility to parse string to wagmi's PayableOverrides.value.Returns BigNumber if its convertible or returns -1 if there is error or invalid string
+ * @param  str - any string
+ * @return {BigNumber | number | undefined}
+ */
+export default function parseTxnValue(str: string): BigNumber | number | undefined {
```

Comment:
### Working : 
- It returns `-1` denoting that the value passed as a string cannot be parsed by BigNumber. 
- Due to this  whenever a user enters eg : `fdsffdfsf` he gets an error displayed as 
> invalid hex string (argument="value", value="-0x01", code=INVALID_ARGUMENT, version=bytes/5.7.0)
In the above error, we can see its telling `value=-0x01` which is kind of misguidance but didn't find a better a way to do it 😅

Also feel free to suggest a better name

Reply:
probably you can use https://docs.ethers.org/v5/api/utils/bignumber/#BigNumberish as a param instead of a string

Reply:
Actually, I wanted to pass values that come from input field that's why kept it string, check https://github.com/scaffold-eth/se-2/pull/96#discussion_r1073185982, but happy to reconsider it :)

Reply:
maybe return undefined instead of -1 and check it before sending tx? Have you tried to send tx with -1 overrides value? Ideally, we need to check the value and if it's wrong show an error in the form and disable the button. Probably we need to create issue for that.

Reply:
https://github.com/scaffold-eth/se-2/pull/101 changed tx value input, so txValue is always correct BigNumber or undefined, so it seems you don't need this function anymore

Reply:
Since merging  https://github.com/scaffold-eth/se-2/pull/104 I think we need to use this util again.

I added it here: https://github.com/scaffold-eth/se-2/pull/96/commits/2b3de3b15e37d5af6f8341b25ed0c42077121e36

I agree that we can handle this better, but let's do it in another issue.